### PR TITLE
create the log file along with its parent directory if not present

### PR DESCRIPTION
**Pull request by bharath-123**
**What type of PR is this?**
Enhancement

**What does this PR do? Why is it needed?**
Currently, when we specify a log file for the beacon chain node and if the log file parent directories are not present then we don't create the directory and error out. This PR adds support to create the parent directories if not present.

**Which issues(s) does this PR fix?**

Fixes (https://github.com/prysmaticlabs/prysm/issues/12496)

**Other notes for review**

### CODE REVIEW

File: io/logs/logutil.go [hunk 1]
```diff
@@ -20,7 +21,10 @@ func addLogWriter(w io.Writer) {
 // ConfigurePersistentLogging adds a log-to-file writer. File content is identical to stdout.
 func ConfigurePersistentLogging(logFileName string) error {
 	logrus.WithField("logFileName", logFileName).Info("Logs will be made persistent")
-	f, err := os.OpenFile(logFileName, os.O_CREATE|os.O_WRONLY|os.O_APPEND, params.BeaconIoConfig().ReadWritePermissions) // #nosec G304
+	if err := os.MkdirAll(filepath.Dir(logFileName), params.BeaconIoConfig().ReadWriteExecutePermissions); err != nil {
+		return err
+	}
+	f, err := os.OpenFile(logFileName, os.O_CREATE|os.O_WRONLY|os.O_APPEND, params.BeaconIoConfig().ReadWriteExecutePermissions) // #nosec G304
```
Comment (prestonvanloon) [inline io/logs/logutil.go:27]: This changes the file permissions. Does the log file need +x?

```suggestion
	f, err := os.OpenFile(logFileName, os.O_CREATE|os.O_WRONLY|os.O_APPEND, params.BeaconIoConfig().ReadWritePermissions) // #nosec G304
```

Comment (bharath-123) [reply io/logs/logutil.go:27]: Apologies @prestonvanloon , will fix this. that was not an intended change. The ReadWriteExecute is mainly for the directory to create a file within it.

Comment (bharath-123) [reply io/logs/logutil.go:27]: the log file doesnt require +x

Comment (bharath-123) [reply io/logs/logutil.go]: should be resolved now

File: io/logs/logutil.go [hunk 2]
```diff
@@ -20,6 +21,9 @@ func addLogWriter(w io.Writer) {
 // ConfigurePersistentLogging adds a log-to-file writer. File content is identical to stdout.
 func ConfigurePersistentLogging(logFileName string) error {
 	logrus.WithField("logFileName", logFileName).Info("Logs will be made persistent")
+	if err := os.MkdirAll(filepath.Dir(logFileName), params.BeaconIoConfig().ReadWriteExecutePermissions); err != nil {
```
Comment (nisdas) [inline io/logs/logutil.go:24]: please use from our `io/file` package to create these paths, it is the standard method we use across prysm for our file i/o operations

**Review by nisdas — COMMENTED** [review]
Can you add a regression test for this ?

File: io/logs/logutil_test.go [hunk 3]
```diff
@@ -24,3 +27,58 @@ func TestMaskCredentialsLogging(t *testing.T) {
 		require.Equal(t, MaskCredentialsLogging(test.url), test.maskedUrl)
 	}
 }
+
+func TestConfigurePersistantLogging(t *testing.T) {
```
Comment (nisdas) [inline io/logs/logutil_test.go:31]: For temporary directories for tests use `t.TempDir()`  , this gets cleaned up by the test runner after the test is executed.

**Review by prestonvanloon — DISMISSED** [review]

**Review by prestonvanloon — APPROVED** [review]

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**